homework-9/script-1.js

```diff
@@ -0,0 +1,187 @@
+        Number.isInteger(this.countryPerson) || this.countryPerson === undefined || this.countryPerson === null ||
+        Number.isInteger(this.listCountryForTravelling) || this.listCountryForTravelling === undefined || this.listCountryForTravelling === null) {
+        console.log('Sorry, something went wrong');
+    } else {
+        this.acquaintance();
+        this.goToBed();
+        this.wakeUp();
+        this.travellInCountries();
+    }
+}
+const person1 = new Person('Olga', 29, 'female', 'ukrainian', 'Ukraine', ['Austria', 'Poland', 'Germany']);
+person1.checkInfo();
+
+// 2 variant
+
+function Person(namePerson, agePerson, sexPerson, nationalityPerson, countryPerson, listCountryForTravelling) {
+    this.namePerson = namePerson;
+    this.agePerson = agePerson;
+    this.sexPerson = sexPerson;
+    this.nationalityPerson = nationalityPerson;
+    this.countryPerson = countryPerson;
+    this.listCountryForTravelling = listCountryForTravelling;
+
+}
+const acquaintance = function () {
+
+    console.log(`Hello ${this.namePerson}. Your information: your age is ${this.agePerson}, your sex is ${this.sexPerson}, your nationality is ${this.nationalityPerson}, your country is ${this.countryPerson}`);
+}
+
+
+const wakeUp = function () {
+
+    console.log(`Hello ${this.namePerson}. It's time to wake up!`);
+}
+
+
+const goToBed = function () {
+
+    console.log(`Hello ${this.namePerson}. It's time to go to bed!`);
+}
+
+const travellInCountries = function () {
+
+    console.log(`${this.namePerson} want to travell in such countries as ${this.listCountryForTravelling}`);
+}
+
+const checkInfo = function () {
+    if (Number.isInteger(this.namePerson) || this.namePerson === undefined || this.namePerson === null ||
+        !Number.isInteger(this.agePerson) || this.agePerson < 0 ||
+        Number.isInteger(this.sexPerson) || this.sexPerson === undefined || this.sexPerson === null ||
+        Number.isInteger(this.nationalityPerson) || this.nationalityPerson === undefined || this.nationalityPerson === null ||
+        Number.isInteger(this.countryPerson) || this.countryPerson === undefined || this.countryPerson === null ||
+        Number.isInteger(this.listCountryForTravelling) || this.listCountryForTravelling === undefined || this.listCountryForTravelling === null) {
+        console.log('Sorry, something went wrong');
+    } else {
+        acquaintance();
+        goToBed();
+        wakeUp();
+        travellInCountries();
+    }
+}
+const person2 = new Person('Olga', 29, 'female', 'ukrainian', 'Ukraine', ['Austria', 'Poland', 'Germany']);
+checkInfo.apply(person2);
+
+//Створити власну реалізацію методу .bind (підказка: в кінці треба помістити цю власну 
+//функцію у прототип - Function.prototype.myOwnBind = function () { [тут_код_кастомного_bind] }
+function Dog(nameDog, ageDog, sexDog) {
+    this.nameDog = nameDog;
+    this.ageDog = ageDog;
+    this.sexDog = sexDog;
+}
+const myOwnBind = function () {
```

Comment:
Own Bind should be created as `Function.prototype.myBind = function .....`
This one looks like simple greeting function, that don't have any connection to binding